Yafc.Model/Data/DataClasses.cs

```diff
@@ -556,18 +695,24 @@ public class EntityEnergy {
     public TemperatureRange acceptedTemperature { get; internal set; } = TemperatureRange.Any;
     public float emissions { get; internal set; }
     public float drain { get; internal set; }
-    public float fuelConsumptionLimit { get; internal set; } = float.PositiveInfinity;
+    public float baseFuelConsumptionLimit { get; internal set; } = float.PositiveInfinity;
+    public float FuelConsumptionLimit(Quality quality) => quality.ApplyStandardBonus(baseFuelConsumptionLimit);
     public Goods[] fuels { get; internal set; } = [];
     public float effectivity { get; internal set; } = 1f;
 }
 
 public class ModuleSpecification {
     public string category { get; internal set; } = null!;
-    public float consumption { get; internal set; }
-    public float speed { get; internal set; }
-    public float productivity { get; internal set; }
-    public float pollution { get; internal set; }
-    public float quality { get; internal set; }
+    public float baseConsumption { get; internal set; }
+    public float baseSpeed { get; internal set; }
+    public float baseProductivity { get; internal set; }
+    public float basePollution { get; internal set; }
+    public float baseQuality { get; internal set; }
+    public float Consumption(Quality quality) => baseConsumption >= 0 ? baseConsumption : quality.ApplyModuleBonus(baseConsumption);
+    public float Speed(Quality quality) => baseSpeed <= 0 ? baseSpeed : quality.ApplyModuleBonus(baseSpeed);
+    public float Productivity(Quality quality) => baseProductivity <= 0 ? baseProductivity : quality.ApplyModuleBonus(baseProductivity);
+    public float Pollution(Quality quality) => basePollution >= 0 ? basePollution : quality.ApplyModuleBonus(basePollution);
+    public float Quality(Quality quality) => baseQuality <= 0 ? baseQuality : quality.ApplyModuleBonus(baseQuality);
```

Comment:
Just to be sure:
Some of these use `>=` and other `<=`, it feels (intuitively) wrong. Is this indeed as intended?

Reply:
Yes, that's intentional. Beneficial effects are boosted, while detrimental ones are not. The code here checks for detrimental conditions (negative speed, prod, and quality; positive consumption and pollution) and returns the unmodified value in those cases.